File: lib/java/src/main/java/uapi/MockUApiSchema.java
package uapi;

import static uapi.internal.schema.CreateMockUApiSchemaFromJsonDocuments.createMockUApiSchemaFromFileJsonMap;
import static uapi.internal.schema.GetSchemaFileMap.getSchemaFileMap;

import java.util.List;
import java.util.Map;

import uapi.internal.types.UFieldDeclaration;
import uapi.internal.types.UType;

/**
 * A parsed uAPI schema.
 */
public class MockUApiSchema {

    public final List<Object> original;
    public final Map<String, UType> parsed;
    public final Map<String, UFieldDeclaration> parsedRequestHeaders;
    public final Map<String, UFieldDeclaration> parsedResponseHeaders;

    public MockUApiSchema(List<Object> original,
            Map<String, UType> parsed,
            Map<String, UFieldDeclaration> parsedRequestHeaders,
            Map<String, UFieldDeclaration> parsedResponseHeaders) {
        this.original = original;
        this.parsed = parsed;
        this.parsedRequestHeaders = parsedRequestHeaders;
        this.parsedResponseHeaders = parsedResponseHeaders;
    }

    public static MockUApiSchema fromJson(String json) {
        return createMockUApiSchemaFromFileJsonMap(Map.of("auto_", json));
    }

    public static MockUApiSchema fromFileJsonMap(Map<String, String> fileJsonMap) {
        return createMockUApiSchemaFromFileJsonMap(fileJsonMap);
    }

    public static MockUApiSchema fromDirectory(String directory) {
        final var map = getSchemaFileMap(directory);
        return createMockUApiSchemaFromFileJsonMap(map);
    }
}

File: lib/py/uapi/MockUApiSchema.py
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from uapi.internal.types.UFieldDeclaration import UFieldDeclaration
    from uapi.internal.types.UType import UType


class MockUApiSchema:
    """
    A parsed uAPI schema.
    """

    def __init__(self, original: list[object], parsed: dict[str, 'UType'], parsed_request_headers: dict[str, 'UFieldDeclaration'],
                 parsed_response_headers: dict[str, 'UFieldDeclaration']):
        self.original = original
        self.parsed = parsed
        self.parsed_request_headers = parsed_request_headers
        self.parsed_response_headers = parsed_response_headers

    @staticmethod
    def from_json(json: str) -> 'MockUApiSchema':
        from uapi.internal.schema.CreateMockUApiSchemaFromFileJsonMap import create_mock_uapi_schema_from_file_json_map
        return create_mock_uapi_schema_from_file_json_map({"auto_": json})

    @staticmethod
    def from_file_json_map(file_json_map: dict[str, str]) -> 'MockUApiSchema':
        from uapi.internal.schema.CreateMockUApiSchemaFromFileJsonMap import create_mock_uapi_schema_from_file_json_map
        return create_mock_uapi_schema_from_file_json_map(file_json_map)

    @staticmethod
    def from_directory(directory: str) -> 'MockUApiSchema':
        from uapi.internal.schema.CreateMockUApiSchemaFromFileJsonMap import create_mock_uapi_schema_from_file_json_map
        from uapi.internal.schema.GetSchemaFileMap import get_schema_file_map
        schema_file_map = get_schema_file_map(directory)
        return create_mock_uapi_schema_from_file_json_map(schema_file_map)
